kernel/src/actions/set_transaction.rs

```diff
@@ -64,8 +63,8 @@ impl SetTransactionScanner {
         )
     }
 
-    /// Scan the Delta Log for the latest transaction entry of an application
-    pub(crate) fn application_transaction(
+    /// Scan the Delta Log for the latest `txn` action for an application id
+    pub(crate) fn latest_txn(
```

Comment:
We don't use `get_XXX` for field getters in rust, but if you squint this is an associative container of sorts, and those _do_ have `get` methods (because you're requesting a specific key from the container):
```suggestion
    pub(crate) fn get_one(
```

We also already know this is about transaction ids because of the class, no need to say it again.

Finally, I don't think we need to talk about "latest" because log replay _always_ takes the latest version of everything. Unless we plan to add an API for fetching a not-latest version?